pkg/controller/datadogagentdeployment/utils.go

```diff
@@ -50,38 +85,171 @@ func newAgentPodTemplate(logger logr.Logger, agentdeployment *datadoghqv1alpha1.
 		},
 		Spec: corev1.PodSpec{
 			ServiceAccountName: getAgentServiceAccount(agentdeployment),
-			InitContainers:     getInitContainers(logger, agentdeployment),
-			Containers: []corev1.Container{
-				{
-					Name:            "agent",
-					Image:           spec.Agent.Image.Name,
-					ImagePullPolicy: *spec.Agent.Image.PullPolicy,
-					Command: []string{
-						"agent",
-						"start",
-					},
-					Resources: *spec.Agent.Config.Resources,
-					Ports: []corev1.ContainerPort{
-						{
-							ContainerPort: 8125,
-							Name:          "dogstatsdport",
-							Protocol:      "UDP",
-						},
-					},
-					Env:           getEnvVarsForAgent(logger, agentdeployment),
-					VolumeMounts:  getVolumeMountsForAgent(spec),
-					LivenessProbe: getDefaultLivenessProbe(),
-				},
+			InitContainers:     initContainers,
+			Containers:         containers,
+			Volumes:            getVolumesForAgent(spec),
+			Tolerations:        agentdeployment.Spec.Agent.Config.Tolerations,
+		},
+	}, nil
+}
+
+func isAPMEnabled(dad *datadoghqv1alpha1.DatadogAgentDeployment) bool {
+	if dad.Spec.Agent == nil {
+		return false
+	}
+	return datadoghqv1alpha1.BoolValue(dad.Spec.Agent.Apm.Enabled)
+}
+
+func isProcessEnabled(dad *datadoghqv1alpha1.DatadogAgentDeployment) bool {
+	if dad.Spec.Agent == nil {
+		return false
+	}
+	return datadoghqv1alpha1.BoolValue(dad.Spec.Agent.Process.Enabled)
+}
+
+func isSystemProbeEnabled(dad *datadoghqv1alpha1.DatadogAgentDeployment) bool {
+	if dad.Spec.Agent == nil {
+		return false
+	}
+	return datadoghqv1alpha1.BoolValue(dad.Spec.Agent.SystemProbe.Enabled)
+}
+
+func getAgentContainers(dad *datadoghqv1alpha1.DatadogAgentDeployment) (*corev1.Container, error) {
```

Comment:
nit (since it returns 1 single container)
```suggestion
func getAgentContainer(dad *datadoghqv1alpha1.DatadogAgentDeployment) (*corev1.Container, error) {
```